keps/sig-network/20190324-remove-kube-proxy-autocleanup.md

```diff
@@ -22,59 +22,73 @@ superseded-by:
 
 ## Summary
 
-Remove the automatic network rule cleanup from kube-proxy.
-Only clean up rules from running in other proxy modes when using the `--cleanup` command.
+Remove any clean up functionality in kube-proxy that is done automatically/implicitly.
+More concretely, kube-proxy should only attempt to clean up its proxy rules when the `--cleanup` flag is set.
```

Comment:
The original is more precise.  We might still clean up rules left behind by kube-proxy IN THE SAME MODE - because we are smart enough to know the delta in rules.  It's not tenable between proxier modules, so that's the real change here.

Reply:
Thanks I'll follow-up. I'm curious about when we would want to clean up proxy rules in the same mode during startup that we can't do during the usual iptables/IPVS sync. That seems unlikely to me but maybe I'm missing something here.